In [8]:
from math import sqrt
import numpy as np
import scipy.stats as stats

# Домашняя работа по уроку 6. Сравнение долей. Построение доверительного интервала

-------------------

## Задача 1

Известно, что генеральная совокупность распределена нормально

со средним квадратическим отклонением, равным 16.

Найти доверительный интервал для оценки математического ожидания с надежностью 0.95,

если выборочная средняя M = 80, а объем выборки n = 256.

### Решение

Так как интервал будем строить для оценки математического ожидания

и нам известно среднее квадратическое отклонение, используем формулу:

X - Z<sub>α/2</sub> * (σ / sqrt(n))

X + Z<sub>α/2</sub> * (σ / sqrt(n))

In [6]:
std = 16
alpha = 0.05
x = 80
n = 256
min = x - 1.96 * (std / sqrt(n))
max = x + 1.96 * (std / sqrt(n))
min, max

(78.04, 81.96)

Ответ: p(78.04 < x < 81.96) = 0.95

-------------------

## Задача 2

В результате 10 независимых измерений некоторой величины X, 

выполненных с одинаковой точностью, получены опытные данные: 

6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1

Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,

оценить истинное значение величины X при помощи доверительного интервала,

покрывающего это значение с доверительной вероятностью 0,95.

### Решение

Так как оцениваем значение средней, при этом не зная дисперсию, используем формулу:

X - t<sub>α/2</sub> * (σ / sqrt(n))

X + t<sub>α/2</sub> * (σ / sqrt(n))

In [11]:
alpha = 0.05
x = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
n = 10
mean = np.mean(x) # среднее
t = stats.t.ppf(0.975, 9) # t-критерий
std = np.std(x, ddof=1) # среднее квадратичное отклонение несмещенное 
min = mean - t * (std / sqrt(n)) # нижняя граница интервала
max = mean + t * (std / sqrt(n)) # верхняя граница интервала
mean, min, max

(6.590000000000001, 6.267515851415713, 6.912484148584288)

Ответ: p(6.27 < x < 6.91) = 0.95

-----------

## Задача 3

Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170

Рост матерей 178, 165, 165, 173, 168, 155, 160, 164, 178, 175

Используя эти данные, построить 95% доверительный интервал 

для разности среднего роста родителей и детей.

### Решение

Интервальная оценка для разности средних арифметических:

delta - t<sub>α/2</sub> * S<sub>delta</sub>

delta + t<sub>α/2</sub> * S<sub>delta</sub>

delta = x1 - x2

D = 1/2 * (D1 + D2)

S<sub>delta</sub> = sqrt(D/n1 + D/n2)

df = 2 * (n - 1)

In [14]:
x_1 = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])
x_2 = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])
n1 = len(x_1)
n2 = len(x_2)
# delta
mean_1 = np.mean(x_1)
mean_2 = np.mean(x_2)
delta = mean_1 - mean_2
# t
t = stats.t.ppf(0.975, 2 * (n1 - 1))
# D
var_1 = np.var(x_1)
var_2 = np.var(x_2)
D = 1/2 * (var_1 + var_2)
# S
S = np.sqrt(D/n1 + D/n2)

min = delta - t * S
max = delta + t * S

mean_1, mean_2, delta, min, max

(166.2, 168.1, -1.9000000000000057, -9.649241760831261, 5.84924176083125)

Ответ: p(-9.65 < delta < 5.85) = 0.95